mirrord/layer/src/socket/ops.rs

```diff
@@ -275,6 +274,22 @@ pub(super) fn bind(
         Err(HookError::AddressAlreadyBound(requested_address))?;
     }
 
+    #[cfg(target_os = "macos")]
+    unsafe {
+        let experimental = crate::setup().experimental();
+        if experimental.disable_reuseaddr {
+            tracing::debug!("here!!");
+            let socket = socket2::Socket::from_raw_fd(sockfd);
+            if let Err(e) = socket.set_reuse_address(false) {
+                tracing::debug!(?e, "Failed to set SO_REUSEADDR to false");
+            }
+            if let Err(e) = socket.set_reuse_port(false) {
+                tracing::debug!(?e, "Failed to set SE_REUSEPORT to false");
+            }
+            // get ownership of fd back
+            let _ = socket.into_raw_fd();
```

Comment:
I'd use [nix function](https://docs.rs/nix/latest/nix/sys/socket/fn.setsockopt.html) and `BorrowedFd::borrow_raw` for this. Just a suggestion tho

Reply:
nice!